src/index.js

```diff
@@ -6,33 +6,32 @@ const pkg = require('../package.json');
 
 const ACTION_UA = `${pkg.name}/${pkg.version}`;
 
-// Sets the required env info for Percy to work correctly
-function setPercyBranchBuildInfo(pullRequestNumber) {
-  if (!!github.context.payload) {
-    return;
-  }
-
-  if (!!pullRequestNumber) {
-    let prBranch = github.context.payload.pull_request.head.ref;
-
-    core.exportVariable('PERCY_BRANCH', prBranch);
-    core.exportVariable('PERCY_PULL_REQUEST', pullRequestNumber);
-  } else {
-    core.exportVariable('PERCY_BRANCH', github.context.payload.ref.replace('refs/heads/', ''));
-  }
-}
-
 (async () => {
   try {
     let flags = core.getInput('exec-flags');
     let testCommand = core.getInput('command');
     let customCommand = core.getInput('custom-command');
     let isDebug = core.getInput('verbose') === 'true';
+    let isPassthough = core.getInput('passthrough') === 'true';
     let isSilenced = core.getInput('silence') === 'true';
     let workingDir = core.getInput('working-directory');
     let pullRequestNumber = github.context.payload.number;
     let execOptions = { cwd: workingDir };
 
+    // Sets the required env info for Percy to work correctly
+    function setPercyBranchBuildInfo() {
+      if (!!pullRequestNumber) {
+        let prBranch = github.context.payload.pull_request.head.ref;
+
+        core.exportVariable('PERCY_BRANCH', prBranch);
+        core.exportVariable('PERCY_PULL_REQUEST', pullRequestNumber);
+      } else if (github.context.payload.ref) {
+        core.exportVariable('PERCY_BRANCH', github.context.payload.ref.replace('refs/heads/', ''));
+      } else {
+        if (isDebug) console.log('Could not set `PERCY_BRANCH`');
+      }
+    }
```

Comment:
Could this function still be declared where it was, but add a second debug arg?

Also, rather than `else { if () ...` why not just `else if () { ...`?